plankton/adapters/__init__.py

```
@@ -152,13 +166,19 @@ def __set__(self, instance, value):
         :param instance: Instance of type.
         :param value: Type.
         """
-        setattr(getattr(instance, self._target_member), self._prop, value)
+        if instance is not None:
+            setattr(getattr(instance, self._target_member), self._prop, value)
 
 
 class ForwardMethod(object):
     def __init__(self, target, method):
         self._target = target
         self._method = method
 
+        try:
+            self.__doc__ = getattr(self._target, self._method).__doc__
+        except AttributeError:
+            pass
```

Is this try catch necessary?

Assuming this is for the case that `_method` has no docstring... doesn't `__doc__` default to `None`, making the assignment succeed?

And if `_method` doesn't actually exist, shouldn't this be treated as an error rather than silently ignored?

---

I looked at it and tried to remember a reason why I did that, but couldn't find any. It was probably a "carry-over" from `ForwardProperty` (where it also turned out to be unnecessary). In `StreamAdapter` this can never happen anyway, because an error is raised before `ForwardMethod` is constructed.

And you're right, it's no problem if this is an error already at this point because everything else in this will fail in that case...